connection.js

```diff
@@ -478,34 +460,41 @@ class Connection {
             return genericAllowedFamiliesReject("timecontrolsunranked", notification.time_control.time_control);
         }
 
-        const resultMaintime = UHMAEAT("maintime", notification.time_control, notification.ranked);
+        return { reject: false }; // OK !
+
+    } /* }}} */
+    // Check challenge settings are allowed
+    //
+    checkChallengeSettings(notification) { /* {{{ */
+
+        let handicapSettings =  {notif: notification.handicap, isFakeHandicap: false};
+        if (notification.handicap === -1 && config.fakerank) {
+            // TODO : modify or remove fakerank code whenever server sends us automatic handicap 
+            //        notification.handicap different from -1.
+            /* adding a .floor : 5.9k (6k) vs 6.1k (7k) is 0.2 rank difference,
+            /  but it is still a 6k vs 7k = 1 rank difference = 1 automatic handicap stone*/
+            handicapSettings = { notif: Math.abs(Math.floor(notification.user.ranking) - Math.floor(config.fakerank)),
+                                 isFakeHandicap: true };
+        }
+        const resultHandicap = minmaxHandicapRankRejectResult("handicap", handicapSettings.notif, handicapSettings.isFakeHandicap, notification.ranked);
+        if (resultHandicap) {
+            return (resultHandicap);
+        }
+        const resultMaintime = UHMAEATRejectResult("maintime", notification.time_control, notification.ranked);
         if (resultMaintime) {
             return (resultMaintime);
         }
-        const resultPeriods = minmaxPeriodsActualReject("periods", notification.time_control.periods, notification.time_control.speed, notification.ranked);
+        const resultPeriods = minmaxPeriodsRejectResult("periods", notification.time_control.periods, notification.time_control.speed, notification.ranked);
```

Comment:
Shouldnt this be `minMaxPeriodsRejectResult`?

Reply:
(i mean, shouldnt we rename it to use proper cammel case?)

Reply:
do we consider `minmax` to be one word, or `minMax` to be 2 words, for someone who doesnt know anything about our code, it seems `minMax` conveys better what we want i think actually indeed